src/lib/Components/ArtworkGrids/GenericGrid.tsx

```diff
@@ -66,9 +66,9 @@ class GenericArtworksGrid extends React.Component<Props, State> {
     }
   }
 
-  shouldComponentUpdate(_, nextState) {
-    if (this.state.sectionCount === nextState.sectionCount) {
-      // if there's a change in columns, we'll need to re-render
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
+    // if there's a change in columns, we'll need to re-render
+    if (this.props.artworks === nextProps.artworks && this.state.sectionCount === nextState.sectionCount) {
```

Comment:
This makes sense - the grid should re-render when artworks change. We briefly discussed offline about potential performance implications, particularly around `InfiniteScrollGrid` and Relay pagination containers, and whether new pages would trigger re-renders of grids backing earlier pages.

Reply:
Looks like the InfiniteScrollGrid doesn't use the GenericGrid at all too